crates/client/aptos/src/client.rs

```diff
@@ -0,0 +1,85 @@
+use std::sync::Arc;
+
+use aptos_sdk::rest_client::Client;
+use aptos_sdk::types::account_address::AccountAddress;
+use aptos_sdk::types::LocalAccount;
+use dc_metrics::{Gauge, MetricsRegistry, PrometheusError, F64};
+use starknet_types_core::felt::Felt;
+use url::Url;
+
+use crate::utils::{new_account, view_request};
+
+#[derive(Clone, Debug)]
+pub struct L1BlockMetrics {
+    pub l1_block_number: Gauge<F64>,
+    pub l1_gas_price_wei: Gauge<F64>,
+    pub l1_gas_price_strk: Gauge<F64>,
+}
+
+impl L1BlockMetrics {
+    pub fn register(registry: &MetricsRegistry) -> Result<Self, PrometheusError> {
+        Ok(Self {
+            l1_block_number: registry
+                .register(Gauge::new("deoxys_l1_block_number", "Gauge for deoxys L1 block number")?)?,
+
+            l1_gas_price_wei: registry.register(Gauge::new("deoxys_l1_gas_price", "Gauge for deoxys L1 gas price")?)?,
+            l1_gas_price_strk: registry
+                .register(Gauge::new("deoxys_l1_gas_price_strk", "Gauge for deoxys L1 gas price in strk")?)?,
+        })
+    }
+}
+
+pub struct AptosClient {
+    pub provider: Client,
+    pub l1_core_contract: Arc<LocalAccount>,
+    pub l1_block_metrics: L1BlockMetrics,
+}
+
+impl Clone for AptosClient {
+    fn clone(&self) -> Self {
+        AptosClient {
+            provider: self.provider.clone(),
+            l1_core_contract: Arc::clone(&self.l1_core_contract),
+            l1_block_metrics: self.l1_block_metrics.clone(),
+        }
+    }
+}
+
+impl AptosClient {
```

Comment:
maybe we should define trait for swapping out between many types of Client easier